sympy/core/basic.py

```diff
@@ -1525,6 +1525,20 @@ def match(self, pattern, old=False):
         pattern = sympify(pattern)
         return pattern.matches(self, old=old)
 
+
+    def match_between(self, pattern1, pattern2, old=False):
```

Comment:
Yah! sure, `match_between` just for the time being compares two patterns but later I plan to compare all possible set with the `self`